### PR TITLE
enable change batch job priority

### PR DESCRIPTION
**Pull request by luispinedamx**
- Applying a general solution to methods returning void
- Line 1342 of SchedulerServiceImpl is a rollback from the previous PR

### CODE REVIEW

**Comment by Mykhailenko** [comment]
retest this please

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java [hunk 1]
```diff
@@ -1362,6 +1361,31 @@ private int executeFunction(BiFunction<RestClient, Integer, InputStream> action,
         return success;
     }
 
+    private int executeVoidFunction(BiFunction<RestClient, Integer, Void> action, List<Integer> jobIdList,
+            String actionName) throws ServiceException, RestServerException {
+
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java [hunk 2]
```diff
@@ -1362,6 +1361,31 @@ private int executeFunction(BiFunction<RestClient, Integer, InputStream> action,
...
+        RestClient restClientProxy = getRestClientProxy();
+
+        int failures = 0;
+        int success = 0;
+
+        for (Integer jobId : jobIdList) {
+            try {
+                action.apply(restClientProxy, jobId);
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MAJOR Extract this nested try block into a separate method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java [hunk 3]
```diff
@@ -444,12 +444,11 @@ public InputStream apply(RestClient restClientProxy, Integer jobId) {
     @Override
     public void setPriorityByName(final String sessionId, List<Integer> jobIdList, final String priorityName)
             throws ServiceException, RestServerException {
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java [hunk 4]
```diff
@@ -444,12 +444,11 @@ public InputStream apply(RestClient restClientProxy, Integer jobId) {
...
-        executeFunction(new BiFunction<RestClient, Integer, InputStream>() {
+        executeVoidFunction(new BiFunction<RestClient, Integer, Void>() {
```
Comment (sonarqube-activeeon) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java]: MAJOR Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 126 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 41 major
* ![MINOR][MINOR] 81 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [SchedulerServiceImpl.java#L279](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L279): Use try-with-resources or close this "JarFile" in a "finally" clause. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)
1. ![CRITICAL][CRITICAL] [SchedulerServiceImpl.java#L165](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L165): Define a constant instead of duplicating this literal "sessionId" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [SchedulerServiceImpl.java#L172](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L172): Define a constant instead of duplicating this literal "ISO-8859-1" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [SchedulerServiceImpl.java#L188](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L188): Define a constant instead of duplicating this literal "Failed to read response: " 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [RestClient.java#L290](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L290): Method has 10 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RestClient.java#L322](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/RestClient.java#L322): Method has 11 parameters, which is greater than 7 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L192](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L192): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L235](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L235): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L272](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L272): Use "java.nio.Files#delete" here for better messages on error conditions. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)
1. ![MAJOR][MAJOR] [SchedulerServiceImpl.java#L369](https://github.com/ow2-proactive/scheduling-portal/blob/40a9e44114fe117139997f1150e627a98d6eff5f/scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SchedulerServiceImpl.java#L369): Make this anonymous inner class a lambda [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**